catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloadManager.java

```diff
@@ -108,40 +111,47 @@ public ResourceResponse download(ResourceRequest resourceRequest, Metacard metac
+            resource = downloaderConfig.getResourceCache()
                     .getValid(new CacheKey(metacard, resourceRequest).generateKey(), metacard);
-            if (cachedResource != null) {
+            if (resource != null) {
                 resourceResponse = new ResourceResponseImpl(resourceRequest,
                         resourceRequest.getProperties(),
-                        cachedResource);
+                        resource);
                 LOGGER.debug("Successfully retrieved product from cache for metacard ID = {}",
                         metacard.getId());
+                return resourceResponse;
             } else {
                 LOGGER.debug("Unable to get resource from cache. Have to retrieve it from source");
             }
         }
 
-        if (resourceResponse == null) {
-            try {
-                resourceResponse = retriever.retrieveResource();
-            } catch (ResourceNotFoundException | ResourceNotSupportedException | IOException e) {
-                throw new DownloadException("Cannot download resource", e);
-            }
-
-            resourceResponse.getProperties()
-                    .put(Metacard.ID, metacard.getId());
-            // Sources do not create ResourceResponses with the original ResourceRequest, hence
-            // it is added here because it will be needed for caching
-            resourceResponse = new ResourceResponseImpl(resourceRequest,
-                    resourceResponse.getProperties(),
-                    resourceResponse.getResource());
-
-            resourceResponse = startDownload(downloadIdentifier,
-                    resourceResponse,
-                    retriever,
-                    metacard);
-
+        try {
+            resourceResponse = retriever.retrieveResource();
+            resource = resourceResponse.getResource();
+        } catch (ResourceNotFoundException | ResourceNotSupportedException | IOException e) {
+            throw new DownloadException("Cannot download resource", e);
         }
+
+        resourceResponse.getProperties()
+                .put(Metacard.ID, metacard.getId());
+        // Sources do not create ResourceResponses with the original ResourceRequest, hence
+        // it is added here because it will be needed for caching
+        resourceResponse = new ResourceResponseImpl(resourceRequest,
+                resourceResponse.getProperties(),
+                resourceResponse.getResource());
+
+        // TODO - this should be before retrieveResource() but eventPublisher requires a
+        // resourceResponse and that resource response must have a resource request in it (to get
+        // USER property)
+        publishStartEvent(metacard, resourceResponse, downloadIdentifier);
+
+        resourceResponse = startDownload(metacard,
+                retriever,
+                resourceResponse,
+                downloadIdentifier,
+                managerReliableResource,
+                doCaching,
```

Comment:
`doCaching` is initialized to `false` on L99 and never updated so either we're missing something or it's not needed anymore.